hypothesis-python/src/hypothesis/searchstrategy/strategies.py

```diff
@@ -435,12 +435,16 @@ def __init__(self, strategies, bias=None):
         self.__element_strategies = None
         self.bias = bias
         self.__in_branches = False
-        if bias is not None:
-            assert 0 < bias < 1
-            self.sampler = cu.Sampler(
-                [bias ** i for i in range(len(strategies))])
-        else:
-            self.sampler = None
+        self.__sampler = None
+
+    @property
+    def sampler(self):
+        if self.__sampler is None:
+            assert 0 < self.bias < 1
```

Comment:
nit: It might be worth having an `assert self.bias is not None` before this. This may fail confusingly, especially under Python 2, otherwise.